src/Service/Payment.php

```diff
@@ -497,4 +507,30 @@ public function isPdfSession(): bool
     {
         return (bool) Registry::getRequest()->getRequestParameter('pdfConfirm', '0');
     }
+
+    /**
+     * @param string $paymentId
+     * @param string $customerType
+     * @param string $currency
+     * @return null|\UnzerSDK\Resources\Payment
+     */
+    private function getFetchPayment(string $paymentId, string $customerType, string $currency)
+    {
+        $tmpFetchPayment = oxNew(TmpFetchPayment::class);
+        $result = $tmpFetchPayment->loadFetchPayment($paymentId);
```

Comment:
this hint might be to complex but at least in theory this can lead to a performance issue depending on how many times this getFetchPayment is called. Might be more performant to add a middle layer like memcached or redis?